pkg/proxy/proxy.go

```diff
@@ -117,13 +118,20 @@ type Proxy struct {
 	endpointClients  []*EndpointClient
 	stripeAuthClient *stripeauth.Client
 	webSocketClient  *websocket.Client
+	isConnected      chan struct{}
 
 	// Events is the supported event types for the command
 	events map[string]bool
 }
 
 const maxConnectAttempts = 3
 
+// IsConnected signals a WaitGroup when the proxy has finished connecting
+func (p *Proxy) IsConnected(wg *sync.WaitGroup) {
+	<-p.isConnected
+	wg.Done()
+}
+
```

Comment:
the latest commit now does this, but in order to keep modifications to Proxy to a minimum, I had to add one inefficiency: looping until `p.webSocketClient` exists, because we can't call `p.webSocketClient.Connected()` until it is initialized, which happens in the `p.Run` connect for loop

one other approach was to use a channel between `p.Run` and `p.IsConnected` but this would likely require extra state in Proxy which I think was mentioned to be unideal